### DISCUSSION TITLE
How can I contribute to the codebase?

### DISCUSSION THREAD

**Discussion by loongx2** [infiniflow/infinity#2302] [General]
Hi how do I contribute to this project ?

**Comment by yingfeng** [comment]
You could take a look at the roadmap #338 at first , and see what kind of features you would like to contribute.

**Comment by JinHai-CN** [comment]
> Hi how do I contribute to this project ?

1. Try to install and run the test cases of Infinity.
2. Try to write some test cases, such as we can write more examples to help user to use Infinity easily.
3. Implement some functions. You can check #2033
4. As Yingfeng said, #338 is the ROADMAP.
